src/core/gui/PageView.cpp

```diff
@@ -793,10 +793,10 @@ double XojPageView::getWidth() const { return page->getWidth(); }
 double XojPageView::getHeight() const { return page->getHeight(); }
 
 void XojPageView::rerenderRect(double x, double y, double width, double height) {
-    int rx = std::lround(std::max(x - 10, 0.0));
-    int ry = std::lround(std::max(y - 10, 0.0));
-    int rwidth = std::lround(width + 20);
-    int rheight = std::lround(height + 20);
+    int rx = std::lround(std::max(x, 0.0));
+    int ry = std::lround(std::max(y, 0.0));
+    int rwidth = std::lround(width);
+    int rheight = std::lround(height);
```

Comment:
This whole rectangle preprocessing is useless, I think. I think `rerenderRect` should just pass its arguments to `addRenderRect` and not do anything else.
RenderJob::rerenderRect handles the double coordinates properly (calling std::floor and std::ceil as it should), so there is not need for any of this.
```suggestion
```